### PR TITLE
CLOUD-2186 Upgrade actions-python to use ubuntu-latest

### PR DESCRIPTION
**Pull request by vibindaniel**
# Description

Updated to a different type of actions.

Fixes [#2186](https://usxpress.atlassian.net/browse/CLOUD-2186)

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Tested in demo-python-flash-api
https://github.com/variant-inc/demo-python-flask-variant-api/actions/runs/4939725042/jobs/8831073524

- [x] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added documentation to test

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - master
+      - main
+
+jobs:
+  Release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Create Tag
+        uses: actions/github-script@v5
```
Comment (zumic96) [inline .github/workflows/release.yml:17]: Would it make sense to switch this to new actions-collection/create-release-tags action?

File: action.yml [hunk 2]
```diff
@@ -1,26 +1,58 @@
...
+      if: ${{ steps.sonar-setup.outputs.sonar_skip != 'True' }}
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Build and Push Image
+      uses: variant-inc/actions-collection/build-push-image@v2
+      if: ${{ inputs.container_push_enabled == 'true' }}
+      with:
+        dockerfile_dir_path: ${{ inputs.dockerfile_dir_path }}
+        ecr_repository: ${{ inputs.ecr_repository }}
```
Comment (zumic96) [inline action.yml:58]: Missing aws_region input.

**Review by zumic96 — CHANGES_REQUESTED** [review]

File: action.yml [hunk 3]
```diff
@@ -1,26 +1,58 @@
...
+    description: >
+      The python-version input is optional.
+      If not supplied, the action will try to resolve the
+      version from the default `.python-version` file.
+      If the `.python-version` file doesn't exist Python or PyPy
+      version from the PATH will be used.
+      The default version of Python or PyPy in PATH varies
+      between runners and can be changed unexpectedly
+      so we recommend always setting Python version explicitly
+      using the python-version or python-version-file inputs.
```
Comment (cameran20) [inline action.yml:28]: should remove the reference for the python-version-file input because it's a separate input in actions/setup-python

**Review by cameran20 — CHANGES_REQUESTED** [review]

**Comment by vibindaniel** [comment]
@cameran20 @zumic96 done with the changes

**Review by zumic96 — DISMISSED** [review]

**Review by cameran20 — DISMISSED** [review]

File: test/action.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    required: false
+runs:
+  using: composite
+  steps:
+    - name: Find Python Packager
+      id: packager
+      shell: bash
+      run: |
+        echo "packager=pip" >> $GITHUB_OUTPUT
+        if [ -f "requirements.txt" ]; then
```
Comment (zumic96) [inline test/action.yml:32]: Isn't this redundant? As in line above it's already set as pip.

Comment (vibindaniel) [reply test/action.yml]: removed

**Review by cameran20 — APPROVED** [review]

**Pull request merged**